components/mosart/src/riverroute/MOSART_physics_mod.F90

```diff
@@ -628,8 +628,7 @@ subroutine mainchannelRouting(iunit, nt, theDeltaT)
     else if(Tctl%RoutingMethod == 2) then
        call Routing_DW(iunit, nt, theDeltaT)
     else
-       print*, "Please check the routing method! There are only 2 methods available. 1==KW, 2==DW."
-	   call shr_sys_abort('Error in selecting routing method!')
+	   call shr_sys_abort('Wrong routing method! There are only 2 methods available. 1==KW, 2==DW.')
```

Comment:
I am not sure whether the endrun() function (see the example below) is available for MOSART. But we usually use it to terminate the program. It may be just a wrapper of shr_sys_abort but I guess it will be safer and more compatible for parallel computing.

call endrun(msg=' ERROR: CH4 Conservation Error in LakeBGCDynamics during diffusion'//&
                     errMsg(__FILE__, __LINE__))